core/src/main/java/de/bwaldvogel/mongo/backend/Cursor.java

```diff
@@ -0,0 +1,64 @@
+package de.bwaldvogel.mongo.backend;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.ThreadLocalRandom;
+
+import de.bwaldvogel.mongo.bson.Document;
+
+public class Cursor implements Iterable<Document> {
+
```

Comment:
Can an empty cursor be shared across all collections ? IE can (empty) cursor be independent of a collection or it is always attached to a collection ? 

If so, can it be a singleton ?

Reply:
So far each cursor lives inside one collection.
I what case will we need cursors outside collections ?